File: 922-possible-bipartition/possible-bipartition.cpp
class Solution {
public:
    bool possibleBipartition(int n, vector<vector<int>>& dislikes) {
        vector<vector<int>> v(n + 1);
        vector<int> color(n + 1, -1);
        for (auto& i : dislikes) {
            v[i[0]].push_back(i[1]);
v[i[1]].push_back(i[0]);
        }
        queue<int> q;

        for (int i = 1; i <= n; i++) {
            if(color[i]==-1){
                color[i]=0;
            q.push(i);
            while (!q.empty()) {
                int p = q.front();
                q.pop();
                for (auto& u : v[p]) {
                    if (color[u] == -1) {
                        color[u] = 1 - color[p];
                        q.push(u);
                    } else if (color[u] == color[p]) {
                        return false;
                    }
                }
            }
            }
        }
        return true;
    }
};

File: 922-possible-bipartition/possible-bipartition.java
class Solution {
    private boolean check(ArrayList<ArrayList<Integer>> adj, int[] color,int node)
    {
        for(int i : adj.get(node))
        {
            if(color[i]==-1)
            {
                color[i]=1-color[node];
                if(check(adj,color,i))
                {
                    return true;
                }
            }
            else if(color[i]==color[node])
            {
                return true;
            }
        }
        return false;
    }
    public boolean possibleBipartition(int n, int[][] dislikes) {
        ArrayList<ArrayList<Integer>> adj = new ArrayList<>();
        for(int i=0;i<n;i++)
        {
            adj.add(new ArrayList<>());
        }
        for(int i=0;i<dislikes.length;i++)
        {
            int a=dislikes[i][0]-1;
            int b=dislikes[i][1]-1;
            adj.get(a).add(b);
            adj.get(b).add(a);
        }
        int[] color = new int[n];
        Arrays.fill(color,-1);
        for(int i=0;i<n;i++)
        {
            if(color[i]==-1)
            {
                color[i]=1;
                if(check(adj,color,i))
                {
                    return false;
                }
            }
        }
        return true;
    }
}